src/models/monster/types.d.ts

```diff
@@ -29,6 +29,51 @@ type Action = {
   action_options: Choice;
 };
 
+type ArmorClass =
+  | ArmorClassDex
+  | ArmorClassNatural
+  | ArmorClassArmor
+  | ArmorClassSpell
+  | ArmorClassCondition;
+
+type ArmorClassDex = {
+  _id?: boolean;
+  type: 'dex';
+  value: number;
+  desc?: string;
+};
+
+type ArmorClassNatural = {
+  _id?: boolean;
+  type: 'natural';
+  value: number;
+  desc?: string;
+};
+
+type ArmorClassArmor = {
+  _id?: boolean;
+  type: 'armor';
+  value: number;
+  armor?: APIReference[]; // Equipment
+  desc?: string;
+};
+
+type ArmorClassSpell = {
+  _id?: boolean;
+  type: 'spell';
+  value: number;
+  spell: APIReference; // Spell
+  desc?: string;
+};
+
+type ArmorClassCondition = {
+  _id?: boolean;
+  type: 'condition';
+  value: number;
+  condition: APIReference; // Condition
+  desc?: string;
+};
+
```

Comment:
I'm curious why you want the armor classes to have IDs. If that's the route you want to go, I think you'll have to use the `ObjectID` type from mongoose. Setting `_id` to false on the schema means the schema won't automatically give the subdocument an ID in mongo, but it doesn't mean the field's type is boolean.

Reply:
The type is not for the fields type but for the Schema's type. Since `_id` is being set to `false`, the type needs to be set to `boolean` or TS gets mad. It's the same thing I have do everywhere else.